photon-lib/src/main/scala/com/linkedin/photon/ml/util/Linalg.scala

```diff
@@ -55,13 +55,38 @@ object Linalg {
     DenseVector(bArr)
   }
 
+  /**
+   * Inverts the matrix $A$, given the Cholesky factorization $l$ of $A$.
+   *
+   * @param l the Cholesky factorization of matrix A
+   * @return the solution
+   */
+  protected[ml] def choleskyInverse(
+      l: DenseMatrix[Double]): DenseMatrix[Double] = {
+
+    val n = l.rows
+
+    // Lapack will write the result to this array
+    val lArr = l.toArray
+
+    val info = new intW(0)
+    lapack.dpotri("L", n, lArr, n, info)
+
+    if (info.`val` < 0) {
+      throw new RuntimeException(
+        s"The value at position ${-info.`val`} had an illegal value.")
+    }
```

Comment:
Please add some comments to this function. What is the `dpotri` call? What is `info`? What is `'val'`? What is an illegal value in this context?

Reply:
I think what I'll do instead is refer to LAPACK documentation, besides a brief description of `dpotri`. It's really their job to document these things.

Reply:
Fair enough